app/workers/insert_lab_records_worker.rb

```diff
@@ -1,22 +1,18 @@
 class InsertLabRecordsWorker
   include Sidekiq::Worker
 
-  def perform(lab_record_import_id, lab_records)
+  def perform(lab_record_import_id)
     Rails.logger.info lab_record_import_id
     @lab_record_import_id = lab_record_import_id
-    @lab_records = lab_records
     import_lab_records
     AnonymizeLabRecordImportWorker.perform_async(lab_record_import.id)
   end
 
   private
 
-  def parsed_records
-    JSON[@lab_records]
-  end
-
-  def import_lab_records
-    parsed_records.each do |lab_record_attributes|
+  def import_lab_records # rubocop:disable Metrics/AbcSize
+    lab_records_attributes = JSON.parse(lab_record_import.lab_records_attributes_file.download)
```

Comment:
Loading the JSON file into memory could be avoided by using `JSON.load` instead and making ActiveStorage to put the blob in a temp file:
```suggestion
    lab_records_attributes = lab_record_import.lab_records_attributes_file.open { |file| JSON.load(file) }
```

Reply:
Unfortunately, `ActiveStorage::Blob -> open` isn't available in our current rails version `5.2.0` (https://apidock.com/rails/v6.0.0/ActiveStorage/Blob/open). I ended up using `download` along with a block, taking the docs into account (https://apidock.com/rails/v5.2.3/ActiveStorage/Blob/download)

Reply:
I'm still struggling to understand the differences between `download` and `open` in terms of memory consumption, though. I mean, what would be the difference between doing:

`myVar = someAttachment.download`

and

`myVar = someAttachment.open`

I mean, wouldn't the result be the same? That is, the whole content of the file loaded into `myVar` and therefore in memory.
Or maybe it's that when using `open` the required memory is freed immediately after reading the content of the file, whereas when using `download` that memory remains allocated until the end of the execution of that function?

Reply:
mm... I don't think that will work for large files. According to the docs: "then the download is streamed and yielded in chunks". So you don't get the full contents at once.

Reply:
So maybe the best thing to do is going back to `download` without block approach?

Reply:
(To complete the story: We ended up using Kernel#open instead of download)